src/Analyzer.Cli/Loggers/SummaryLogger.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Templates.Analyzer.Cli
+{
+    /// <summary>
+    /// Class to print a summary of the execution logs
+    /// </summary>
+    public class SummaryLogger : ILogger
+    {
+        private readonly Dictionary<string, int> loggedErrors = new();
+        private readonly Dictionary<string, int> loggedWarnings = new();
+
+        /// <summary>
+        /// Constructor of the SummaryLogger class
+        /// </summary>
+        public SummaryLogger()
+        {
+        }
+
+        /// <inheritdoc/>
+        public IDisposable BeginScope<TState>(TState state) => default!;
+
+        /// <inheritdoc/>
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            var recordLog = new Action<Dictionary<string, int>, string>((logs, newLogMessage) => {
+                if (!logs.TryAdd(newLogMessage, 1))
+                {
+                    logs[newLogMessage] += 1;
+                }
+            });
+
+            var logMessage = state.ToString();
+
+            if (logLevel == LogLevel.Error)
+            {
+                recordLog(loggedErrors, logMessage);
+            }
+            else if (logLevel == LogLevel.Warning)
+            {
+                recordLog(loggedWarnings, logMessage);
+            }
+        }
+
+        /// <summary>
+        /// Outputs to console a summary of the errors and warnings logged
+        /// </summary>
+        public void SummarizeLogs(bool showVerboseMessage)
+        {
+            if (loggedErrors.Count > 0 || loggedWarnings.Count > 0)
+            {
+                Console.ForegroundColor = loggedErrors.Count > 0 ? ConsoleColor.Red : ConsoleColor.Yellow;
+
+                Console.WriteLine($"{Environment.NewLine}{loggedErrors.Count} error(s) and {loggedWarnings.Count} warning(s) were found during the execution, please refer to the original messages above");
```

Comment:
I am not sure I agree with the color choice here, but don't want to make it more complex and also have to consider accessibility and color contrast.
- Thinking in red is the error part and in yellow is the warning part and in default is the rest.
- (note for myself) Want to explore and test out this experience first

Reply:
And question, the "original messages above", what color are those? Isn't that the section below so should this say below?

Reply:
(I did not test this locally so sorry about that).

Reply:
I'd prefer limiting the red/yellow colors to just the error/warning messages.  But there is also value in making the summary as a whole stand out somehow.  I'm not sure how best to do that.

Reply:
Actually, regarding `"please refer to the original messages above"` - this summary is going to be printed even when SARIF is specified as the report format, and it will be more confusing.  This should be updated to take that into account.

Reply:
Errors and warnings are shown in the console even when the report format chosen was SARIF, so that message would still be accurate

Reply:
No, errors and warnings are printed to console when they happen, and this summary is printed at the end of the execution

Reply:
My suggestion would be to include the report format selected by the user when creating this logger.  Then:
- If console, the messaging can stay as-is
- If SARIF, just print the *total* numbers of errors and warnings, and instruct to refer to the tool notifications in the SARIF file for details.

Reply:
So, we don't want any console logging when using SARIF?